src/NexusMods.App.UI/WorkspaceSystem/GridUtils.cs

```diff
@@ -438,94 +438,245 @@ internal static bool IsPerfectGrid(WorkspaceGridState gridState)
         return res.UnionById(updates);
     }
 
-    internal static IReadOnlyList<ResizerInfo> GetResizers(
-        ImmutableDictionary<PanelId, Rect> currentState,
-        bool isWorkspaceHorizontal = true)
+    internal static List<ResizerInfo> GetResizers2(WorkspaceGridState workspaceState)
     {
-        var tmp = new List<ResizerInfo>(capacity: currentState.Count);
-        var adjacentPanels = new List<ResizerInfo>(capacity: currentState.Count);
+        return workspaceState.IsHorizontal
+            ? GetResizersForHorizontal(workspaceState)
+            : GetResizersForVertical(workspaceState);
+    }
+
+    private static List<ResizerInfo> GetResizersForVertical(WorkspaceGridState currentState)
```

Comment:
Could you add some comments here and there, to make the code easier to read. 
These are long functions and following what is going on gets hard.